src/semgrep_agent/semgrep.py

```diff
@@ -274,7 +293,8 @@ def invoke_semgrep(
             for c in chunk:
                 args.append(c)
 
-            _ = semgrep_exec(*args, _timeout=timeout, _err=debug_echo)
+            exit_code = semgrep_exec(*args, _timeout=timeout, _err=debug_echo).exit_code
+            max_exit_code = max(max_exit_code, exit_code)
```

Comment:
I would also document this "max of all" behavior with a comment where exit_code is defined at the top of FindingSets